### PR TITLE
DDF-3307 Move test third-party libraries so they can be re-used outside integration tests

### PR DESCRIPTION
**Pull request by lessarderic**
#### What does this PR do?
Moves DDF versions of Restito and RestAssured currently under `distribution/test/thirdparty` to the top-level directory `thirdparty` and changes maven coordinates accordingly.

This will allow other tests, such as component tests, to be able to use them without creating a circular dependency.

#### Who is reviewing it?
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@brendan-hofmann 
@tbatie 
@AzGoalie 

#### Select relevant component teams: 
@codice/test 

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl
@rzwiefel

#### How should this be tested? (List steps with links to updated documentation)
Full build successful.

#### Any background context you want to provide?
Pre-requisite to https://github.com/codice/ddf/pull/2090.

#### What are the relevant tickets?
[DDF-3307](https://codice.atlassian.net/browse/DDF-3307)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [X] Update / Add Integration Tests

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by AzGoalie** [comment]
Are we removing `SecureStubServer.java` because it's not used?

**Comment by lessarderic** [comment]
Good catch @AzGoalie! The file must have been removed from the git index when I rebased and I didn't think to check and add it back. I'll do that right now.

**Comment by lessarderic** [comment]
build now

**Comment by brendan-hofmann** [comment]
Failure looks legit: `20:21:55 [ERROR] Child module /jenkins/workspace/DDF-Jobs/pr/Linux/thirdparty/pom.xml of /jenkins/workspace/DDF-Jobs/pr/Linux/pom.xml does not exist @ 
`

**Comment by lessarderic** [comment]
build now

**Comment by lessarderic** [comment]
build now

**Comment by lessarderic** [comment]
build now

**Comment by lessarderic** [comment]
build now

File: thirdparty/pom.xml [hunk 1]
```diff
@@ -17,17 +17,20 @@
...
-        <artifactId>test</artifactId>
+        <groupId>ddf</groupId>
+        <artifactId>ddf</artifactId>
         <version>2.11.0-SNAPSHOT</version>
     </parent>
-    <groupId>ddf.test.thirdparty</groupId>
+
+    <groupId>ddf.thirdparty</groupId>
     <artifactId>thirdparty</artifactId>
     <name>DDF :: Test :: Thirdparty</name>
```
Comment (tbatie) [inline thirdparty/pom.xml]: Should we update this name?

File: thirdparty/restito/src/main/java/com/xebialabs/restito/server/secure/SecureStubServer.java [hunk 2]
```diff
@@ -51,9 +51,6 @@
 @SuppressFBWarnings("MF_CLASS_MASKS_FIELD")
 public class SecureStubServer extends StubServer {
 
-  @SuppressWarnings("WeakerAccess")
-  public static final int DEFAULT_PORT = 6666;
```
Comment (rzwiefel) [inline thirdparty/restito/src/main/java/com/xebialabs/restito/server/secure/SecureStubServer.java]: ❓ Were these two fields already declared in the parent class, so we could remove them?

**Comment by rzwiefel** [comment]
@lessarderic Have we tried running the itests locally to see if we are getting the same failures as CI?

File: thirdparty/pom.xml (hunk 1 above)
Comment (lessarderic) [reply thirdparty/pom.xml]: Good catch! Done.

File: thirdparty/restito/src/main/java/com/xebialabs/restito/server/secure/SecureStubServer.java (hunk 2 above)
Comment (lessarderic) [reply thirdparty/restito/src/main/java/com/xebialabs/restito/server/secure/SecureStubServer.java]: Done.

**Comment by lessarderic** [comment]
build now

**Comment by lessarderic** [comment]
build now

**Comment by lessarderic** [comment]
build now

**Pull request merged**